core/lib/dal/src/data_availability_dal.rs

```diff
@@ -0,0 +1,220 @@
+        }
+        Ok(())
+    }
+
+    /// Saves the inclusion data for the given L1 batch. If the inclusion data is already present,
+    /// verifies that it matches the one provided in the function arguments
+    /// (meaning that the inclusion data corresponds to the same DA blob)
+    pub async fn save_l1_batch_inclusion_data(
+        &mut self,
+        number: L1BatchNumber,
+        da_inclusion_data: &[u8],
+    ) -> DalResult<()> {
+        let update_result = sqlx::query!(
+            r#"
+            UPDATE data_availability
+            SET
+                inclusion_data = $1,
+                updated_at = NOW()
+            WHERE
+                l1_batch_number = $2
+                AND inclusion_data IS NULL
+            "#,
+            da_inclusion_data,
+            i64::from(number.0),
+        )
+        .instrument("save_l1_batch_da_data")
+        .with_arg("number", &number)
+        .report_latency()
+        .execute(self.storage)
+        .await?;
+
+        if update_result.rows_affected() == 0 {
+            tracing::debug!("L1 batch #{number}: DA data wasn't updated as it's already present or the row for the batch_number is missing");
+
+            let instrumentation =
+                Instrumented::new("get_matching_batch_da_data").with_arg("number", &number);
+
+            // Batch was already processed. Verify that existing DA data matches
+            let query = sqlx::query!(
+                r#"
+                SELECT
+                    COUNT(*) AS "count!"
+                FROM
+                    data_availability
+                WHERE
+                    l1_batch_number = $1
+                    AND inclusion_data = $2
+                "#,
+                i64::from(number.0),
+                da_inclusion_data,
+            );
+
+            let matched: i64 = instrumentation
+                .clone()
+                .with(query)
+                .report_latency()
+                .fetch_one(self.storage)
+                .await?
+                .count;
+
+            if matched != 1 {
+                let err = instrumentation.constraint_error(anyhow::anyhow!(
+                        "DA data verification failed. DA data for L1 batch #{number} does not match the one provided before"
+                    ));
+                return Err(err);
+            }
+        }
+        Ok(())
+    }
+
+    /// Assumes that the L1 batches are sorted by number, and returns the first one that is ready for DA dispatch.
+    pub async fn get_first_da_blob_awaiting_inclusion(
+        &mut self,
+    ) -> DalResult<Option<DataAvailabilityBlob>> {
+        Ok(sqlx::query_as!(
+            StorageDABlob,
+            r#"
+            SELECT
+                l1_batch_number,
+                blob_id,
+                inclusion_data,
+                sent_at
+            FROM
+                data_availability
+            WHERE
+                inclusion_data IS NULL
+                AND blob_id IS NOT NULL
```

Comment:
yes, otherwise there is a chance we will try to check for the inclusion of the blobs that weren't even dispatched.
with the current logic it's not possible, but it's an easy precaution if we decide to parallelize the `da_dispatcher`'s logic in the future